index.js

```diff
@@ -212,24 +241,58 @@ NYC.prototype.report = function (cb, _collector, _reporter) {
   reporter.write(collector, true, cb)
 }
 
+var loadedMaps = {}
+
 NYC.prototype._loadReports = function () {
   var _this = this
   var files = fs.readdirSync(this.tmpDirectory())
 
-  return _.map(files, function (f) {
+  var SourceMapCache = require('./lib/source-map-cache')
+  var sourceMapCache = new SourceMapCache()
+
+  var cacheDir = _this.cacheDirectory()
+
+  return files.map(function (f) {
+    var report
     try {
-      return JSON.parse(fs.readFileSync(
+      report = JSON.parse(fs.readFileSync(
         path.resolve(_this.tmpDirectory(), './', f),
         'utf-8'
       ))
     } catch (e) { // handle corrupt JSON output.
       return {}
     }
+
+    if (report) {
+      Object.keys(report).forEach(function (relFile) {
+        var fileReport = report[relFile]
+        if (fileReport && fileReport.contentHash) {
+          var hash = fileReport.contentHash
+          if (!(loadedMaps[hash] || (loadedMaps[hash] === false))) {
```

Comment:
Oh wait that prevents multiple attempts to read the source map. Maybe rewrite to `if (loadedMaps[hash] !== false && !loadedMaps[hash])` so that's a bit more obvious?